### DISCUSSION TITLE
Popover: Floating arrow on Popover

### DISCUSSION THREAD

**Discussion by karlwir** [tailwindlabs/headlessui#3282] [Ideas]
Hi,

While migrating to 2.0 I realised theres no way to have an arrow pointing to the Popvers anchor.

In 1.x I used Popper.js to place my Popovers and could add an arrow myself with a modifier.

FloatingUi has similar functionality, https://floating-ui.com/docs/floatingarrow

Any plans on making it available in Headless UI?

Cheers!

**Comment by stuikomma** [comment]
The preview image on https://headlessui.com/ even suggests that there is an arrow, but there is no documentation on the topic.

<img width="440" alt="image" src="https://github.com/tailwindlabs/headlessui/assets/2040603/abd5bb04-327e-4c6a-959e-9d7057a87664">

I would appreciate an option in headless UI without adding extra libs.

**Reply by lgtmrobo** [reply]
Any luck?

**Reply by stuikomma** [reply]
No, I haven't attempted a custom implementation, yet.

**Comment by giero** [comment]
Today got the same problem :) I have Tooltip component based on `Popover`. For v1 I used to use https://headlessui-float.vercel.app/react/arrow.html, but now it's no longer supported.

> Headless UI Float only support @headlessui/react v1, in the @headlessui/react v2 will direct using the anchor prop to positioning the Headless UI components, please reference the [Headless UI Menu example](https://github.com/tailwindlabs/headlessui/blob/7be23e5c7e2e71218cdba405ea4f4aca7924acc8/playgrounds/react/pages/menu/menu-with-transition.tsx) to using the transition for menu or other components.

But now I have no idea how to add this arrow properly to v2 :thinking:

**Comment by cheryl-c-tirerack** [comment]
<img width="427" alt="image" src="https://github.com/user-attachments/assets/d56a354a-84e1-4614-9a26-74010b0efe9f" />
So... where's the arrow?

**Reply by stuikomma** [reply]
Are asking what we are calling "arrow" here? It is this caret that points to "Solutions"

![image](https://github.com/user-attachments/assets/6e621922-161e-4317-aa5d-d84bb3d49ec6)

**Comment by dagjomar** [comment]
Update: It seems the arrow is now gone from the documentation pages as well

![Screenshot 2025-03-14 at 16 34 47](https://github.com/user-attachments/assets/38f8f78e-6fc6-41c8-9d77-cbbe2702ba5c)

**Reply by stuikomma** [reply]
I think it never was [here](https://headlessui.com/react/popover), because this is the actual code example running.

But it still is [here](https://headlessui.com/) on the main page.

![image](https://github.com/user-attachments/assets/5e9267d6-dcda-4dce-92d0-9375362789d4)
(screenshot with more context)

**Comment by functasti** [comment]
It would be helpful to have a basic tooltip arrow element included in the library to visually connect the `PopoverPanel` to the `PopoverButton`, similar to how Radix Primitives includes a `<Popover.Arrow />` with their popover component.

**Comment by functasti** [comment]
Meanwhile, for anyone looking for a quick fix without using SVGs, here’s how I tackled it:

Under:

```jsx
<PopoverPanel anchor={{ to: "bottom center", gap: "6px" }}>
```

Add this:

```html
<div className="size-[calc(var(--spacing)*5*1.414/2)] mx-auto grid place-content-center translate-y-1/2">
  <div className="size-5 bg-blue-500 rounded-tl-xs rotate-45"></div>
</div>
```

Replace `bg-blue-500` with your panel background color  to have it blend in seamlessly.
